python/ray/autoscaler/aws/config.py

```diff
@@ -250,7 +265,14 @@ def _configure_security_group(config):
             "IpRanges": [{
                 "CidrIp": "0.0.0.0/0"
             }]
-        }])
+        }]
+        try:
+            IpPermissions.extend(config["provider"]["extra_config"][
+                "security_group"]["IpPermissions"])
+        except KeyError:
+            pass
+
+        security_group.authorize_ingress(IpPermissions=IpPermissions)
```

Comment:
Note that this call will fail (without an error message identifying the offending rule) whenever permissions are specified that either duplicate the above default permissions, or are duplicated within the user-provided permissions in config. We can insulate a user from _most_ of these failures by simply removing duplicate IpPermissions via naive equality checks. However, one caveat is that automatic server-side coercion to equivalent, but unequal, representations of the same rule will still slip by (e.g. try injecting [{'FromPort': 22, 'ToPort': 22, 'IpProtocol': 'tcp', 'Ipv6Ranges': [{'CidrIpv6': '2002:0:0:1234::/64'}, {'CidrIpv6': '2002::1234:0:0:0:0/64'}]}] and verify that this call fails due to rule duplication).

Reply:
since this isn't a regression, I'm going to resolve this conversation and address it in some other PR.